### ISSUE TITLE
Passing flags to kustomize

### ISSUE THREAD

**Issue by daddz** [GoogleContainerTools/skaffold#2488]
### Expected behavior
I would like to pass a flag to the `kustomize build` call that skaffold uses but I can't find an option to do so.
The flag I try to pass in this case is `--load_restrictor none`.

### Actual behavior
Existing flags in `skaffold.yaml` are ignored when calling `skaffold dev/deploy`
Log: `DEBU[0001] Running command: [kustomize build ./k8s/test/] `
### Information

- Skaffold version: v0.33.0
- Operating system: Linux
- Contents of skaffold.yaml:

```yaml
deploy:
  kustomize:
    path: ./k8s/test/
    flags:
      global:
        - "--load_restrictor none"
```

**Comment by dgageot** [comment]
Actually, the flags are passed to `kubectl`. Passing flags to kustomize would be a new thing

**Comment by priyawadhwa** [comment]
Hey @daddz thanks for opening this issue. We probably don't have the bandwidth to get to this for a while, but if you or anyone else is interested in opening a PR we would be happy to take a look.

We would need to add a `KustomizeFlags` field to the kustomize config [struct](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/schema/latest/config.go#L410) and then we could pass in those flags when calling kustomize [here](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/deploy/kustomize.go#L250).

```golang
type KustomizeFlags struct {
   // Build are additonal flags passed to kustomize build
   Build []string `yaml:"build,omitempty"`
}
```

**Comment by daddz** [comment]
Hey @priyawadhwa, thanks for the suggestions. I'll take a look at it!

**Comment by dgageot** [comment]
This was fixed with #2870 and #2871
